jetty-ee9/jetty-ee9-security/src/main/java/org/eclipse/jetty/ee9/security/authentication/SslClientCertAuthenticator.java

```diff
@@ -63,14 +64,27 @@ public Authentication validateRequest(ServletRequest req, ServletResponse res, b
 
         HttpServletRequest request = (HttpServletRequest)req;
         HttpServletResponse response = (HttpServletResponse)res;
-        X509Certificate[] certs = (X509Certificate[])request.getAttribute(SecureRequestCustomizer.PEER_CERTIFICATES_ATTRIBUTE);
+        //X509Certificate[] certs = (X509Certificate[])request.getAttribute(SecureRequestCustomizer.PEER_CERTIFICATES_ATTRIBUTE);
+        SecureRequestCustomizer.SslSessionData sslSessionData = (SecureRequestCustomizer.SslSessionData)req.getAttribute(SecureRequestCustomizer.DEFAULT_SSL_SESSION_DATA_ATTRIBUTE);
+
+        if (sslSessionData == null)
+        {
+            return Authentication.SEND_FAILURE;
+        }
+
+        if (sslSessionData.cipherSuite() != null)
+        {
+            req.setAttribute("jakarta.servlet.request.cipher_suite", sslSessionData.cipherSuite());
+        }
+        req.setAttribute("jakarta.servlet.request.key_size", sslSessionData.keySize());
+        X509Certificate[] certs = sslSessionData.peerCertificates();
 
         try
         {
             // Need certificates.
             if (certs != null && certs.length > 0)
             {
-
+                req.setAttribute("jakarta.servlet.request.X509Certificate", certs);
```

Comment:
Same as for EE10. This class should be unchanged, only `SecureRequestCustomizer` should be changed.

Reply:
Do we have a ee10 specific `SecureRequestCustomizer` ?

Reply:
We do not, but we should at this point.